src/main/java/edu/harvard/iq/dataverse/api/Datasets.java

```diff
@@ -4588,7 +4595,12 @@ public Response getPidGenerator(@Context ContainerRequestContext crc, @PathParam
         } catch (WrappedResponse ex) {
             return error(Response.Status.NOT_FOUND, "No such dataset");
         }
-        String pidGeneratorId = dataset.getPidGeneratorId();
+        PidProvider pidProvider = dataset.getEffectivePidGenerator();
+        if(pidProvider == null) {
+            //This is basically a config error, e.g. if a valid pid provider was removed after this dataset used it
+            return error(Response.Status.NOT_FOUND, "No PID Generator found for the give id");
```

Comment:
nitpick here - should be "given" and maybe put it in the bundle. Also should we add a test to PidUtilTest?

Reply:
done